rust-runtime/aws-smithy-http-server/src/lib.rs

```diff
@@ -11,25 +11,28 @@
 pub(crate) mod macros;
 
 pub mod body;
-pub mod error;
-mod extension;
+pub(crate) mod error;
 pub mod routing;
 
+#[doc(hidden)]
+pub mod extension;
```

Comment:
The extension module can be used by external entities such as a Tower metrics layer, so I believe it should be public and with docs as it will be used outside of the codegen.